backends/arm/arm_partitioner.py

```diff
@@ -100,7 +96,7 @@ def is_node_supported_custom(self, node: torch.fx.Node) -> bool:
 @final
 class ArmPartitioner(Partitioner):
     def __init__(self, compile_spec: List[CompileSpec]) -> None:
-        self.delegation_spec = DelegationSpec(ArmBackend.__name__, compile_spec)
+        self.delegation_spec = ArmBackendSelector.get_delegation_spec(compile_spec)
```

Comment:
> I like this, that said what is the rationale for keeping the partitioner class shared?

The idea is to keep the ArmPartitioner as a generic entrypoint for TOSA backends. Both to share code and to separate the Ethos-U backend from the TOSA backend. But not using the TOSA name since that name maybe isn't super clear to users.

>We should try to avoid making quantizer aware of backend specific details by passing in some flags, because then it will be a clunky UX. If we find a need for that then we should just have wrappers.

I do think the quantizer also needs to have some flags to know how to quantize according to the TOSA specification the hardware supports. What is it that you think get's clunky passing it in to a generic TOSA/Arm quantizer?
Not as part of this PR, so let's take that discussion of line.

Reply:
From, UX side I think it is much cleaner if you have a different front ends for each TOSA compliant hardware backend i.e. using ethos and vulkan as proxy tosa backends for an example,

Option-A

```
q = VulkanQuantizer()
...
em = to_backend(VulkanPartitioner(vulkan_flag=1))
...
vulkan_runtime.c

--- different user ---

q = EthosQuantizer()
...
em = to_backend(EthosPartitioner(ethos_flag=1))
...
ethos_runtime.c
```

Option-B
```
q = ArmQuantizer(flag=Vulkan)
...
em = to_backend(ArmPartitioner(flag=Vulkan, vulkan_flag=1))
...
vulkan_runtime.c

--- different user ---

q = ArmQuantizer(flag=Ethos)
...
em = to_backend(ArmPartitioner(flag=Ethos, ethos_flag=1))
...
ethos_runtime.c
```

Purely from UX point of view I like Option-A, I see that as namespaces or inheritance of classes. TOSA is the backbone, which we want to hide from people (per your comment, which I agree) can be behind the scene as an implementation detail.

Also Option-A allows you to do more vulkan or ethos specific things without messing with the other i.e. if you want to add vela specific flags, it just a noise for Vulkan users.
Moreover, the overlap between the set of users is not much i.e. there is no reason for Ethos users to be "aware" of Vulkan specific things or even the existence of Vulkan for that matter.

On the flip side, if Quantizer and Partitioners are both targeting TOSA spec then you can argue that there might not be much specific to different TOSA backend. I agree to an extend, however there will always be things like hardware or runtime related creeps in.